### PR TITLE
Update mip_test.c

### PR DESCRIPTION
**Pull request by jfsimon1981**
HTTP fetch using MIP stack.

### CODE REVIEW

**Comment by jfsimon1981** [comment]
HTTP fetch with MIP stack.
Compilation warnings (void* to int conversion).
Additional Makefile WARN flags -Wno-pointer-to-int-cast -Wno-int-to-pointer-cast

File: test/mip_test.c [hunk 1]
```diff
@@ -38,6 +54,8 @@ static void test_queue(void) {
   assert(q_avail(&q) == 0);
 }
 
+// *****************************************************************************
```
Comment (cpq) [inline test/mip_test.c:57]: Please remove this.
And below

File: test/mip_test.c [hunk 2]
```diff
@@ -49,9 +67,108 @@ static void test_statechange(void) {
   onstatechange(&iface);
 }
 
+// *****************************************************************************
+// Implement tap_rx() tap_tx() tap_up()
+
+static size_t tap_rx(void *buf, size_t len, void *userdata) {
+  ssize_t received = read((int) userdata, buf, len); // Original
```
Comment (cpq) [inline test/mip_test.c:74]: As discussed, please try this to squash the warning:

```c
read((int) (size_t) userdata, buf, len);
```

If that does not work, search for other ways on how to do a cast without a warning.

File: test/mip_test.c [hunk 3]
```diff
@@ -49,9 +67,108 @@ static void test_statechange(void) {
...
+  } else if (ev == MG_EV_HTTP_MSG) {
+    struct mg_http_message *hm = (struct mg_http_message *) ev_data;
+    // MG_INFO(("-->[%.*s]", (int) hm->message.len, hm->message.ptr));
+    // ASSERT(mg_vcmp(&hm->method, "HTTP/1.1") == 0);
+    // ASSERT(mg_vcmp(&hm->uri, "301") == 0);
+    *fn_data_int_ptr = mg_http_status(hm);
+  } else if (ev == MG_EV_CLOSE) {
+    if (*fn_data_int_ptr == 0) *fn_data_int_ptr = 888;
+  } else if (ev == MG_EV_ERROR) {
+    if (*fn_data_int_ptr == 0) *fn_data_int_ptr = 777;
```
Comment (cpq) [inline test/mip_test.c:113]: That's a copy/paste from the unit_test which is not used in this test. Please remove EVERYTHING that is not used. Leave only relevant code.

File: test/mip_test.c [hunk 4]
```diff
@@ -49,9 +67,108 @@ static void test_statechange(void) {
...
+  int ok = 0;
+  struct mg_connection *conn;
+
+  int pc = 100; // Wait sufficiently to acquire HTTP lease
+  while ((pc--)>0) {
+    mg_mgr_poll(&mgr, 100);
+    usleep(10000); // 10 ms
+  }
+
+  conn = mg_http_connect(&mgr, "http://cesanta.com", f_http_fetch, &ok);
```
Comment (cpq) [inline test/mip_test.c:153]: We might do several HTTP requests, to fetch small and large chunks, with and without SSL.
Therefore, please make a function `char *fetch(struct mg_mgr *mgr, const char *url, const char *post_data)` that can be called multiple times. The function should return full HTTP message including headers.

**Review by cpq — CHANGES_REQUESTED** [review]

**Pull request closed without merging**